src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,15 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(!fNRows && !fNCols && fNCols!=fNRows )
```

Comment:
This condition is never true - it requires `fNRows = fNCols = 0 && fNRows != fNRows`. Guessing you meant `||`?